test/test_enforcer.cpp

```diff
@@ -40,27 +40,27 @@ namespace test_enforcer
             return filepath1;
         }
 
-        void TestEnforce(Enforcer* e, string sub, string dom, string obj, string act, bool res) {
+        void TestEnforce(shared_ptr<Enforcer> e, string sub, string dom, string obj, string act, bool res) {
             Assert::AreEqual(res, e->Enforce(sub, dom, obj, act));
         }
 
-        void TestEnforce(Enforcer* e, string sub, string obj, string act, bool res) {
+        void TestEnforce(shared_ptr<Enforcer> e, string sub, string obj, string act, bool res) {
             Assert::AreEqual(res, e->Enforce(sub, obj, act));
         }
 
-        void TestEnforce(Enforcer* e, vector<string> params, bool res) {
+        void TestEnforce(shared_ptr<Enforcer> e, vector<string> params, bool res) {
             Assert::AreEqual(res, e->Enforce(params));
         }
 
-        void TestEnforce(Enforcer* e, unordered_map<string,string> params, bool res) {
+        void TestEnforce(shared_ptr<Enforcer> e, unordered_map<string,string> params, bool res) {
             Assert::AreEqual(res, e->Enforce(params));
         }
 
 
         TEST_METHOD(TestFourParams) {
             string model = filePath("../examples/rbac_with_domains_model.conf");
             string policy = filePath("../examples/rbac_with_domains_policy.csv");
-            Enforcer* e = Enforcer::NewEnforcer(model, policy);
+            shared_ptr<Enforcer> e = Enforcer::NewEnforcer(model, policy);
```

Comment:
How about using `auto` so no need to change so much code when you change a return type?